zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/expression/EnvVariableEvaluationContext.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.expression;
+
+import io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter;
+import java.util.List;
+import java.util.Objects;
+import org.agrona.DirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+
+public class EnvVariableEvaluationContext implements ScopedEvaluationContext {
+  public static final List<String> ALLOWED_ENVIRONMENT_VARIABLE_PREFIXES =
+      List.of("CUSTOM_", "ZEEBE_");
+
+  private final List<String> allowedPrefixes;
+
+  public EnvVariableEvaluationContext() {
+    allowedPrefixes = ALLOWED_ENVIRONMENT_VARIABLE_PREFIXES;
+  }
+
+  public EnvVariableEvaluationContext(final List<String> allowedPrefixes) {
+    this.allowedPrefixes = allowedPrefixes;
+  }
+
+  @Override
+  public DirectBuffer getVariable(final String variableName) {
+    Objects.requireNonNull(variableName);
+
+    if (isAllowed(variableName)) {
+      final String value = System.getenv(variableName);
+
+      if (value != null) {
+        return asMsgPack(value);
+      }
+    }
+
+    return null;
+  }
+
+  private static UnsafeBuffer asMsgPack(final String value) {
+    // @TODO is this the best way to convert to MsgPack?
```

Comment:
💭 Not sure, if this is the best way to convert the value of the environment variable to something the feel expression can properly unpack.

[Here](https://github.com/camunda/camunda/blob/ab273e28f9920688886fb53f9cb6c62224aa4d07/zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelExpressionLanguage.java#L49) is defined that the feel engine uses the `MessagePackValueMapper` to resolve variable values. Also the `VariableState` seems to have variables stored using `MsgPack` encoding. But I feel that this connection is quite hidden, so it feels a bit awkward that the `EnvVariableEvaluationContext` needs to know that and is thereby implicitly coupled.